spark3/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -90,7 +96,14 @@ public StructType outputType() {
       if (retainLastNum != null) {
         action.retainLast(retainLastNum);
       }
-
+      if (numThreads != null && numThreads > 0) {
```

Comment:
I'd just default this to 1 instead of null and do a precondition on numThreads > 0

Currently I believe this is just ignoring invalid numThreads right?

Reply:
+1 on adding validation.

I would consider keeping the null value, though, to allow the underlying action to default it to whatever it wants. I think we do that for other params.

Reply:
I think keeping the null value is the correct behavior, because if `executeDeleteWith` is not set, then the execution will happen in the current thread. Otherwise this will create a new thread pool of size 1 and use that for execution.